OpenSim/Common/ComponentConnector.h

```diff
@@ -200,45 +175,110 @@ class OSIMCOMMON_API AbstractConnector : public Object {
+        SimTK_INDEXCHECK_ALWAYS(ix, getNumConnectees(),
+                                "AbstractConnector::getConnecteeName()");
+        return getConnecteeNameProp().getValue(ix);
     }
 
     void appendConnecteeName(const std::string& name) {
         OPENSIM_THROW_IF((getNumConnectees() > 0 && !_isList), Exception,
             "Multiple connectee names can only be appended to a list Connector.");
-        updProperty_connectee_name().appendValue(name);
+        updConnecteeNameProp().appendValue(name);
     }
 
+protected:
+    //--------------------------------------------------------------------------
+    // CONSTRUCTION
+    //--------------------------------------------------------------------------
+    /** Create a Connector with specified name and stage at which it should be
+    connected.
+    @param name               name of the connector, usually describes its dependency.
+    @param connecteeNameIndex Index of the property in the containing Component
+                              that holds this Connector's connectee_name(s).
+    @param connectAtStage     Stage at which Connector should be connected.
+    @param owner              Component to which this Connector belongs. */
+    AbstractConnector(const std::string& name,
+                      const PropertyIndex& connecteeNameIndex,
+                      const SimTK::Stage& connectAtStage,
+                      Component& owner) :
+            _name(name),
+            _connectAtStage(connectAtStage),
+            _connecteeNameIndex(connecteeNameIndex),
+            _owner(&owner),
+            _isList(getConnecteeNameProp().isListProperty()) {}
 
-    /** Disconnect this Connector from its connectee. */
-    virtual void disconnect() = 0;
 
-protected:
     const Component& getOwner() const { return _owner.getRef(); }
-    void setOwner(const Component& o) { _owner.reset(&o); }
-
-private:
-    void constructProperties() {
-        constructProperty_connectee_name();
-        if (!_isList) {
-            updProperty_connectee_name().appendValue("");
-            updProperty_connectee_name().setAllowableListSize(1);
+    /** %Set an internal pointer to the Component that contains this Connector.
+    This should only be called by Component.
+    This exists so that after the containing Component is copied, the 'owner'
+    is the new Component. This Connector needs to be able to modify
+    the associated connectee_name property in the Component. Thus, we require
+    a writeable reference. */
+    // We could avoid the need for this function by writing a custom copy
+    // constructor for Component.
+    void setOwner(Component& o) { _owner.reset(&o); }
+    /** This will be false immediately after copy construction or assignment.*/
+    bool hasOwner() const { return !_owner.empty(); }
+    
+    /** Check if entries of the connectee_name property contain spaces; if so,
+    print out a warning. */
+    void finalizeFromProperties() {
```

Comment:
How about a more specific name like `checkConnecteeNameProperty()`? Keep in mind that this function should disappear if we make the property type into InputConnecteePath or ChannelPath instead of std::string.